### PR TITLE
Add support for Vala-Lint

### PR DESCRIPTION
**Pull request by atsuya**
Hi,

I added support for [Vala-Lint](https://github.com/vala-lang/vala-lint). I've been using it myself, and it's working just fine.

Let me know if you have any questions. Thanks!

### CODE REVIEW

File: ale_linters/vala/vala_lint.vim
Comment (hsanson) [inline ale_linters/vala/vala_lint.vim:10]: I don't think this vala_vala_lint_enable_config variable is needed. We can check if vala_vala_lint_config_filename is empty instead:

```
let l:config_filename = ale#Var(a:buffer, 'vala_vala_lint_config_filename')
let l:config_path = ale#path#FindNearestFile(a:buffer, l:config_filename)  " <- this is empty if l:config_filename is empty

if !empty(l:config_path)
    let l:command .= ' -c ' . l:config_path
endif

return l:command . ' %s'
```

File: ale_linters/vala/vala_lint.vim
Comment (hsanson) [inline ale_linters/vala/vala_lint.vim:60]: Replace with a function call:

```
'executable': function('ale_linters#vala#vala_lint#GetExecutable')
```

**Review by hsanson — CHANGES_REQUESTED** [review]

File: ale_linters/vala/vala_lint.vim
Comment (atsuya) [reply ale_linters/vala/vala_lint.vim]: Applied your suggested changes. Thanks!

File: ale_linters/vala/vala_lint.vim
Comment (atsuya) [reply ale_linters/vala/vala_lint.vim]: The reason for the enable flag was that I thought there might be a case where you want to keep your config filename value in your .vimrc but want to disable the config file search for some machines (i.e. on one machine, the config file is stored in network mounted filesystem, and searching through it can cause a delay).

But I think your suggestion makes sense as it makes things easier. Applied your suggested changes. Thanks!

File: ale_linters/vala/vala_lint.vim
Comment (atsuya) [reply ale_linters/vala/vala_lint.vim]: Applied your suggested changes.

**Comment by atsuya** [comment]
@hsanson Thanks for your review. I applied suggested changes. Please take a look and let me know what you think. Thanks!

**Review by hsanson — DISMISSED** [review]
This is good to merge. Waiting for Travis-CI to get fixed so all checks pass before merging.

**Comment by atsuya** [comment]
Great! I'm holding my 🍺️ until Travis-CI gets fixed.

**Comment by hsanson** [comment]
ALE migrated from Travis to Github Actions (see #3548) so it is necessary to rebase this PR from latest master branch to trigger the required checks.

```
# Add dense-analysis as remote upstream on your local fork. Not 
# needed if already added.
git remote add upstream https://github.com/dense-analysis/ale.git

# Sync your local master with upstream master
git checkout master
git fetch upstream master
git merge upstream/master

# Rebase the PR branch with master
git checkout my-branch-name
git rebase master   # Fix any conflicts you may have

# Force push the updated PR branch to origin to trigger the checks
git push -f origin my-branch-name
```

**Comment by atsuya** [comment]
Done. I also updated the linter name to snake case as the test indicated.

**Review by hsanson — APPROVED** [review]
Good to go, thanks.

**Comment by atsuya** [comment]
Cheers! :beers:

**Pull request merged**
